zap/src/main/java/org/parosproxy/paros/network/HttpMessage.java

```diff
@@ -1046,6 +1046,22 @@ public void mutateHttpMethod(String method) {
 
             getRequestHeader().setMessage(hrh.toString());
             getRequestBody().setBody(body);
+
+            // CONNECT method does not require the entire URI and parses only the authority as the
+            // URI, hence add relevant information
+            if (method.equals(HttpRequestHeader.CONNECT)
+                    || prevMethod.equals(HttpRequestHeader.CONNECT)) {
+                String scheme;
+                if (uri.getScheme() != null) {
+                    scheme = uri.getScheme();
```

Comment:
why change the URI at all in this case?

Reply:
When a CONNECT request is parsed, all information except the scheme is lost. Hence if the original request is CONNECT we have to craft the URI again

Reply:
Note that the scheme should be lost too, the authority is just the host and port

Reply:
the following comment says the opposite of what you're saying here? (talking about line [1058](https://github.com/zaproxy/zaproxy/pull/6464/files#diff-03602e8a54ab912dc5998078c7a4dbde859050d3425c594f50498e810416618cR1058))

Reply:
@thc202 When changing from  CONNECT to anyother method, should I append a `http` by default or just let it be the host and port only

Reply:
Defaulting to `http` seems reasonable, it could also check the port, if it's 443 use `https` otherwise `http`.

Reply:
@thc202 But the `setURI` method of the `HttpRequestHeader` class automatically sets the scheme to `http` if it is null, and thus a URI for a request header cannot be set without the scheme. So should I make changes in `HttpRequestHeader` too?(The URI doesn't seem to have a `setScheme` method to set the scheme afterwords)

Reply:
Yes, the `setURI` would have to take into account the method, like `parse` already does. The scheme would be set to the default value (which is `http`).

Reply:
@thc202 The `URI` object cannot be intialized without a scheme. Thus whenever the method is changed to CONNECT, if I set the scheme to `http` by default as you mention, then the request in the Manual Request Editor will inevitably be shown with the scheme attached;i.e there seems to be no way to not show the scheme in the editor for CONNECT requests without making changes to the models themself. What should I do now?

Reply:
It can, `new URI("example.com:443", true)` works as expected. You can also see CONNECT requests in the Request panel properly (i.e. without scheme).

(The default to `http` was when changing from CONNECT to other method.)

Reply:
@thc202  Well, it seems that initializing `URI` with the default uri without scheme `(www.any_domain_name.org:9000)` throws an incorrect scheme `URIException` whereas it is correctly initialized for `(www.example.com:9000`. This seems to be an effect of the apache commons library.

Reply:
That should be addressed in upstream.